src/components/nabla/common/NablaTokenPrice.tsx

```diff
@@ -1,30 +1,30 @@
-import { NumberLoader } from '../../Loader';
+import Big from 'big.js';
 import { useNablaTokenPrice } from '../../../hooks/nabla/useNablaTokenPrice';
+import { NumberLoader } from '../../Loader';
 
 export type TokenPriceProps = {
   address: string;
   prefix?: ReactNode;
   fallback?: ReactNode;
-  currentTokenAmount?: number;
+  currentTokenAmount?: Big;
   formatByAmount?: boolean;
 };
 
 export function NablaTokenPrice({
-  currentTokenAmount,
   address,
   prefix = null,
   fallback = null,
   formatByAmount = false,
+  currentTokenAmount = Big(0),
 }: TokenPriceProps): JSX.Element | null {
   const { data, isLoading } = useNablaTokenPrice(address);
   if (isLoading) return <NumberLoader />;
 
   if (data === undefined) return fallback;
 
   if (formatByAmount) {
-    const currentAmountPrice = (Number(data.approximateStrings.atLeast2Decimals) * (currentTokenAmount || 0)).toFixed(
-      2,
-    );
+    const currentAmountPrice = Big(data.approximateStrings.atLeast2Decimals).mul(currentTokenAmount).toFixed(2);
```

Comment:
Please write `toFixed(2, 0)`, otherwise it rounds to nearest instead of rounding down.